main.py

```diff
@@ -0,0 +1,160 @@
+"""
+Main functionality for the smart home app.
+"""
+
+import logging
+
+from python.database import AccessUserDatabase
+import hashlib
+import re
+
+log = logging.getLogger(__name__)
+handler = logging.StreamHandler()  # Logs to the terminal
+log.addHandler(handler)  # Adds handler to log
+log.setLevel(logging.INFO)  # Ensure INFO and above are logged
+handler.setLevel(logging.INFO)  # Ensure the handler also logs INFO and above
+
+class SmartHomeApp:
+    def __init__(self):
+        self.database = AccessUserDatabase()
+        self.logged_in_user_id = None
+
+    def hash_password(self, password):
+        """Hashes the password using SHA-256."""
+        hashed_password = hashlib.sha256(password.encode()).hexdigest()
+        return hashed_password
+
+    def verify_username(self, username):
+        if 1 < len(username) < 30:
+            return True
+        log.info("username must be between 1 and 30 characters.")
+        return False
+
+    def verify_email(self, email):
+        if re.fullmatch(r"[^@]+@[^@]+\.[^@]+", email):
+            return True
+        log.info("email did not contain '@' and/or '.'")
```

Comment:
```suggestion
        log.info("email is not in a valid format. ")
```
I don't think we need to be specific about why the email isn't valid, everyone knows what an email address looks like.